recipes/z3/all/conanfile.py

```diff
@@ -48,9 +51,18 @@ def config_options(self):
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
+        if self.options.multiprecision == "internal":
+            self.provides.append("gmp")
 
     def requirements(self):
-        self.requires("mpir/3.0.0")
+        self.output.info(
+            f"{self.name} will build using {self.options.multiprecision} multiprecision implementation.")
+        if self.options.multiprecision == "mpir":
+            self.requires("mpir/3.0.0")
+        elif self.options.multiprecision == "gmp":
+            self.requires("gmp/6.2.1")
+        elif self.options.multiprecision == "internal":
+            pass
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version],
```

Comment:
```suggestion
        get(self, **self.conan_data["sources"][self.version],
```
from conan.tools.files import get

Reply:
Coming from the outside, I find your proposition less understandable, because the word `get` is too versatile.
But done.